src/WorkItemMigrator/JiraExport/JiraItem.cs

```diff
@@ -101,9 +101,14 @@ private static List<JiraRevision> BuildRevisions(JiraItem jiraItem, IJiraProvide
                         if (attachmentChange == null)
                             continue;
 
-                        attachmentChanges.Add(attachmentChange);
-
-                        UndoAttachmentChange(attachmentChange, attachments);
+                        if (UndoAttachmentChange(attachmentChange, attachments))
```

Comment:
It is getting really too if-elsey, not very fond of that. Think it makes sense to move this long if-else thing in the loop to a switch Statement. Also, I would refactor out specific methods foreach step. Like HandleAttahcmentChange-> which will call out attachment-related actions

Reply:
Resolved